src/connect-netlify/netlify-gateway.ts

```diff
@@ -0,0 +1,139 @@
+import { Result, URI } from "@adviser/cement";
+import { bs, rt, getStore, Logger, NotFoundError, SuperThis, ensureSuperLog } from "@fireproof/core";
+import { Base64 } from "js-base64";
+
+export const S3_VERSION = "v0.1-s3";
+
+export interface S3Opts {
+  readonly region: string;
+  readonly accessKeyId: string;
+  readonly secretAccessKey: string;
+}
+
+export class NetlifyGateway implements bs.Gateway {
+  readonly sthis: SuperThis;
+  readonly logger: Logger;
+
+  constructor(sthis: SuperThis) {
+    this.sthis = ensureSuperLog(sthis, "NetlifyGateway");
+    this.logger = this.sthis.logger;
+  }
+
+  buildUrl(baseUrl: URI, key: string): Promise<Result<URI>> {
+    const url = baseUrl.build();
+    url.setParam("key", key);
+    return Promise.resolve(Result.Ok(url.URI()));
+  }
+
+  async destroy(): Promise<Result<void>> {
+    // Implement the destroy logic for Netlify
+    return Result.Ok(undefined);
+  }
+
+  async start(url: URI): Promise<Result<URI>> {
+    await this.sthis.start();
+    this.logger.Debug().Str("url", url.toString()).Msg("start");
+    const ret = url.build().defParam("version", "v0.1-netlify").URI();
+    return Result.Ok(ret);
+  }
+
+  async close(): Promise<bs.VoidResult> {
+    return Result.Ok(undefined);
+  }
+
+  async put(url: URI, body: Uint8Array): Promise<bs.VoidResult> {
+    const { store } = getStore(url, this.sthis, (...args) => args.join("/"));
+    if (store === "meta") {
+      // const jsonBody = JSON.parse(new TextDecoder().decode(body));
+      // console.log({jsonBody});
```

Comment:
these code paths are the same except for this implicit assert (currently commented out)